stan/math/rev/core/operator_unary_not.hpp

```diff
@@ -10,10 +11,14 @@ namespace math {
  * Return the negation of the value of the argument as defined by
  * <code>!</code>.
  *
+ * @tparam A var autodiff type
  * @param[in] x argument
  * @return negation of argument value
  */
-inline bool operator!(const var& x) { return !x.val(); }
+template <typename Var, require_var_t<Var>...>
+inline bool operator!(Var&& x) {
```

Comment:
Because there are no derivatives, this can also be done as a single primitive template using `value_of_rec`.

Reply:
Yep! I think with all these it would be good to do them as a seperate PR else this is going to blast off into like a 800-1000 line change PR